src/Xamarin.Android.Build.Tasks/Tasks/FindLayoutsToBind.cs

```diff
@@ -0,0 +1,79 @@
+using System;
+using System.Collections.Generic;
+using System.IO;
+using System.Linq;
+using System.Text.RegularExpressions;
+
+using Microsoft.Build.Utilities;
+using Microsoft.Build.Framework;
+
+namespace Xamarin.Android.Tasks
+{
+	public class FindLayoutsToBind : Task
+	{
+		static readonly Regex layoutPathRegex = new Regex ($".*{Path.DirectorySeparatorChar}layout(-?\\w+)*{Path.DirectorySeparatorChar}.*\\.a?xml$", RegexOptions.Compiled | RegexOptions.IgnoreCase);
+
+		public bool GenerateLayoutBindings { get; set; }
+
+		public string BindingDependenciesCacheFile { get; set; }
+
+		public ITaskItem[] BoundLayouts { get; set; }
+
+		[Required]
+		public ITaskItem[] ResourceFiles { get; set; }
+
+		[Output]
+		public ITaskItem[] LayoutsToBind { get; set; }
+
+		public override bool Execute ()
+		{
+			Log.LogDebugMessage ("FindLayoutsToBind Task");
+			Log.LogDebugMessage ($"  GenerateLayoutBindings: {GenerateLayoutBindings}");
+			Log.LogDebugMessage ($"  BindingDependenciesCacheFile: {BindingDependenciesCacheFile}");
+			Log.LogDebugTaskItems ("  BoundLayouts:", BoundLayouts);
+			Log.LogDebugTaskItems ("  ResourceFiles:", ResourceFiles);
+
+			var layouts = new Dictionary <string, ITaskItem> (StringComparer.Ordinal);
+			if (GenerateLayoutBindings) {
+				Log.LogDebugMessage ("Collecting all layouts");
+				foreach (ITaskItem item in ResourceFiles) {
+					AddLayoutFile (item.ItemSpec, ref layouts);
```

Comment:
`ref layouts`?! `ref` isn't needed h ere; `AddLayoutFile()` doesn't modify the parameter.

Reply:
yeah, left-over from old code